### ISSUE TITLE
oniguruma dependency broken

### ISSUE THREAD

**Issue by PaulMakepeace** [stedolan/jq#936]
An FYI for anyone else running into this,

```
==> Installing dependencies for jq: oniguruma
==> Installing jq dependency: oniguruma
==> Downloading http://www.geocities.jp/kosako3/oniguruma/archive/onig-5.9.6.tar.gz

curl: (22) The requested URL returned error: 403 Forbidden
Error: Failed to download resource "oniguruma"
Download failed: http://www.geocities.jp/kosako3/oniguruma/archive/onig-5.9.6.tar.gz
```

Which apparently translates to,

> The page can not be displayed
> Sorry, the accessed page is no longer currently available. Please try again later. Please check the help If you have any questions, if that does not work please contact us from within the help page.

**Comment by pkoppstein** [comment]
The .gz can be obtained from:

https://web.archive.org/web/20150803013327/http://www.geocities.jp/kosako3/oniguruma/archive/onig-5.9.6.tar.gz

**Comment by nicowilliams** [comment]
Yeah, we're aware.  We're probably going to include the tarball in the jq repo.  Or perhaps we should have a jq-dependencies repo (which would require attention from @stedolan).  Or perhaps we'll find a sufficiently-official (for some definition of that) successor of Oniguruma that we can use instead (it seems Onigmo would be it).  We'll see.

**Comment by wtlangford** [comment]
I'd be fine with moving to Onigmo, if there wasn't a lack of a presence in
system package managers.  That's my primary concern with the switch.

**Comment by nicowilliams** [comment]
@wtlangford Can you look into whether it's backward compatible?  Do you want to take this issue?

**Comment by wtlangford** [comment]
Yeah, I've got some extra time this weekend so I'll pick it up

**Comment by PaulMakepeace** [comment]
I can confirm tweaking the url in `/usr/local/Library/Formula/oniguruma.rb` to https://web.archive.org/web/20150803013327/http://www.geocities.jp/kosako3/oniguruma/archive/onig-5.9.6.tar.gz and re-running `brew install jq` worked.

Thanks all!

**Comment by nicowilliams** [comment]
@elad: Well, there are several Onigurumas in github.  Which one, if any, is canonical (for some value of canonical)?

**Comment by elad** [comment]
@nicowilliams the one I just pointed you at :) it's by the actual author, K. Kosako (@kkos). 

I emailed him in July about moving it to GitHub so it's not just a tarball on geocities and he kindly agreed. That's why the geocities site is down (https://github.com/kkos/oniguruma/commit/0b58ff7e98fd47fe944bd90e1f53adf65077033e).

**Comment by nicowilliams** [comment]
@elad Aha.  It would have been nice if @kkos had left some evidence of this move in the old geocities site :(  Still, we can check that the 5.9.6 tag on gh matches the 5.9.6 tarball from the web archive of the old geocities site and go from there.  Thanks for the tip.

**Comment by elad** [comment]
If you want to verify project authenticity, the best way would be to email @kkos.

I don't know if the tag matches the old archive, I just added it myself. One of the reasons for the move to GitHub is that development history was not available before that, so I don't know at what point the archive that was on geocities was created.

**Comment by pkoppstein** [comment]
Here is a "diff" of the "old" (onig-5.9.6) and "new" (oniguruma-master at github) repositories.  The files that differ only differ insubstantially (e.g. a new email address).
```
$ diff -rq onig-5.9.6 oniguruma-master
Only in oniguruma-master: .gitignore
Files onig-5.9.6/AUTHORS and oniguruma-master/AUTHORS differ
Only in oniguruma-master: CMakeLists.txt
Files onig-5.9.6/COPYING and oniguruma-master/COPYING differ
Files onig-5.9.6/HISTORY and oniguruma-master/HISTORY differ
Only in onig-5.9.6: Makefile.in
Files onig-5.9.6/README and oniguruma-master/README differ
Files onig-5.9.6/README.ja and oniguruma-master/README.ja differ
Only in oniguruma-master: README.md
Only in onig-5.9.6: aclocal.m4
Only in oniguruma-master: cmake
Only in oniguruma-master: config.h.cmake.in
Only in onig-5.9.6: configure
Only in oniguruma-master: dist.info
diff: oniguruma-master/ltmain.sh: No such file or directory
Only in onig-5.9.6: m4
Files onig-5.9.6/onig-config.in and oniguruma-master/onig-config.in differ
Only in oniguruma-master: oniguruma.pc.cmake.in
Only in oniguruma-master/sample: CMakeLists.txt
Only in onig-5.9.6/sample: Makefile.in
```

**Comment by kmaehashi** [comment]
In [@kkos's blog article](http://d.hatena.ne.jp/kkos/20110703/1309622717), which was [linked from the oniguruma web site](http://web.archive.org/web/20150522232037/http://www.geocities.jp/kosako3/), mentions his GitHub account https://github.com/kkos/.  So I believe that https://github.com/kkos/oniguruma is a genuine repository for Oniguruma.

**Comment by elad** [comment]
This has been fixed in https://github.com/Homebrew/homebrew/commit/5eed615a12a62de9fa6bea599cb9f28bbef1eb0b

**Comment by nicowilliams** [comment]
Thanks @elad and @DomT4!

**Comment by dtolnay** [comment]
Seems like this has been resolved by #937 and https://github.com/Homebrew/homebrew/pull/43699.
